ext/libyajl2/extconf.rb

```diff
@@ -52,6 +56,15 @@ def self.setup_env
       if windows?
         $LDFLAGS << " -Wl,--export-all-symbols -Wl,--enable-auto-import -Wl,--out-implib=libyajldll.a -Wl,--output-def,libyajl.def"
       end
+
+      # The `-shared` flag is required to enable a bunch of shared-library specific behaviours
+      # in the GCC `collect2` wrapper program.
+      #
+      # Ruby's mkmf.rb doesn't pass this flag to GCC when linking on AIX since this commit:
+      # <https://github.com/ruby/ruby/commit/fa0bec050f2743d3e6cdfb1e5fe2c2786ee0c9e>.
+      if aix?
+        $LDFLAGS << " -shared"
```

Comment:
we build omnibus-chef using xlc, not gcc, so likely this would break our AIX builds, and is probably why it was removed awhile back.